src/tests/daos_perf.c

```diff
@@ -105,28 +108,40 @@ ts_abt_init(void)
 		fprintf(stderr, "ABT init failed: %d\n", rc);
 		return -1;
 	}
+
 	rc = ABT_xstream_self(&abt_xstream);
 	if (rc != ABT_SUCCESS) {
 		printf("ABT get self xstream failed: %d\n", rc);
 		return -1;
 	}
+
 	rc = ABT_xstream_get_cpubind(abt_xstream, &cpuid);
 	if (rc != ABT_SUCCESS) {
-		fprintf(stderr, "get cpubind: %d\n", rc);
-		return -1;
+		fprintf(stderr, "get cpubind failed: %d\n", rc);
+		fprintf(stderr, "No CPU affinity for this test.\n");
+		fprintf(stderr, "Build ABT by --enable-affinity if"
+			" you want to try CPU affinity.\n");
+		return 0;
 	}
 
 	rc = ABT_xstream_get_affinity(abt_xstream, 0, NULL,
 				      &num_cpus);
 	if (rc != ABT_SUCCESS) {
 		fprintf(stderr, "get num_cpus: %d\n", rc);
-		return -1;
+		fprintf(stderr, "No CPU affinity for this test.\n");
+		fprintf(stderr, "Build ABT by --enable-affinity if"
+			" you want to try CPU affinity.\n");
+		return 0;
 	}
+
 	cpuid = (cpuid + 1) % num_cpus;
 	rc = ABT_xstream_set_cpubind(abt_xstream, cpuid);
 	if (rc != ABT_SUCCESS) {
 		fprintf(stderr, "set affinity: %d\n", rc);
-		return -1;
+		fprintf(stderr, "No CPU affinity for this test.\n");
+		fprintf(stderr, "Build ABT by --enable-affinity if"
+			" you want to try CPU affinity.\n");
+		return 0;
```

Comment:
Why not using hwloc for CPU affinity? ABT cpu affinity functions are actually noop functions. (it's not enabled by default)

Reply:
Hmm, it needs to create extra thread explicitly if I use hwloc. I can try it later.

Reply:
> Hmm, it needs to create extra thread explicitly if I use hwloc. I can try it later.

I think we need to call ABT_xstream_create_with_rank() to match server behavior anyway (maybe just use ABT pre-defined scheduler), right?

I think it's very important to use same method we used for io server (hwloc) to do CPU binding here, ABT_xstream_set_cpubind() is actually noop operation. I guess we never try CPU binding for daos_perf and compare the performance with server stack, right (could it be the cause of performance gap between daos_perf VOS mode and server stack mode)?